ui/src/app/shared/components/resource-submit.tsx

```diff
@@ -52,7 +52,7 @@ export class ResourceSubmit<T> extends React.Component<ResourceSubmitProps<T>, R
                                     <p />
                                 )}
                                 <button type='button' className='argo-button argo-button--sm' id='uploadWf' onClick={() => document.getElementById('file').click()}>
-                                    Upload Workflow{' '}
+                                    Upload File{' '}
```

Comment:
Hmm... why did you make this change? I fear that saying "Upload File" instead of "Upload Workflow" will make users think they can upload things such as `CronWorkflows` or `WorkflowTemplates` here, which is not true.

Reply:
add resource name.